Add Emitter CRUD Endpoints and Models

snoe opened the pull request:
Resolves https://github.com/metabase/metabase/issues/22851

Defined models and routes for emitter endpoints.

snoe committed:
Add Emitter CRUD Endpoints

Defined models and routs for emitter endpoints.

camsaul commented on src/metabase/cmd/copy.clj:
```diff
@@ -69,6 +70,9 @@
+   CardEmitter
+   DashboardEmitter
+   Emitter
    Activity
```
Seems like it should be included because copying one database to another should copy that table

snoe replied:
done

camsaul commented on src/metabase/api/emitter.clj:
```diff
@@ -0,0 +1,36 @@
+(api/defendpoint DELETE "/:emitter-id"
+  "Endpoint to delete an emitter."
+  [emitter-id]
+  {}
```
```suggestion
```

The schema validations map is optional

camsaul commented on src/metabase/api/emitter.clj:
```diff
@@ -0,0 +1,36 @@
+(api/defendpoint POST "/"
+  "Endpoint to create an emitter."
+  [:as {emitter :body}]
+  {}
```
```suggestion
```

camsaul commented on src/metabase/api/emitter.clj:
```diff
@@ -0,0 +1,36 @@
+(api/defendpoint PUT "/:emitter-id"
+  "Endpoint to update an emitter."
+  [emitter-id :as {emitter :body}]
+  {}
```
```suggestion
```

snoe committed:
Add all new models to cmd Copy

snoe committed:
Update src/metabase/api/emitter.clj

Co-authored-by: Cam Saul <1455846+camsaul@users.noreply.github.com>

snoe committed:
Update src/metabase/api/emitter.clj

Co-authored-by: Cam Saul <1455846+camsaul@users.noreply.github.com>

snoe committed:
Update src/metabase/api/emitter.clj

Co-authored-by: Cam Saul <1455846+camsaul@users.noreply.github.com>

snoe committed:
Fix lints

snoe committed:
Add compojure endpoint for defendpoint

snoe committed:
Don't copy all the new models

In the spirit of rapid prototyping don't worry about supporting copy for
these new models without surrogate :id keys.

https://github.com/metabase/metabase/issues/22934 will cover supporting
this.

snoe committed:
Merge branch 'pod-writeback' into writeback-emitter-endpoints

escherize reviewed (approved):
I'm approving, but please remove the tap> that slipped through

snoe committed:
Update emitter.clj

snoe commented on src/metabase/models/emitter.clj:
```diff
@@ -0,0 +1,97 @@
+                                 (select-keys [:action_id])
+                                 not-empty)]
+    (db/update! EmitterAction (:emitter_id emitter) emitter-action))
+  (tap> emitter)
```
thanks!
